src/components/Admin/Community/LeadingOrganisationPage.tsx

```diff
@@ -0,0 +1,220 @@
+import { useApolloErrorHandler } from '../../../hooks/useApolloErrorHandler';
+import { UpdateChallengeInput } from '../../../types/graphql-schema';
+import Avatar from '../../core/Avatar';
+
+interface Params {
+  ecoverseId: string;
+  challengeId: string;
+}
+
+interface LeadingOrganisationPageProps extends PageProps {}
+
+const LeadingOrganisationPage: FC<LeadingOrganisationPageProps> = ({ paths }) => {
+  const currentPaths = useMemo(() => [...paths, { name: 'lead', real: false }], [paths]);
+  useUpdateNavigation({ currentPaths });
+
+  const handleError = useApolloErrorHandler();
+
+  const { ecoverseId, challengeId: challengeNameId } = useParams<Params>();
+
+  const { data: _challenge } = useChallengeNameQuery({
+    variables: { ecoverseId: ecoverseId, challengeId: challengeNameId },
+  });
+  const challengeId = _challenge?.ecoverse?.challenge.id || '';
+
+  const { data: _organisations } = useOrganisationsListInfoQuery();
+  const organisations = _organisations?.organisations || [];
+
+  const { data: _leadingOrganisations } = useChallengeLeadOrganisationsQuery({
+    variables: { ecoverseId: ecoverseId, challengeID: challengeNameId },
+  });
+  const leadingOrganisations = _leadingOrganisations?.ecoverse.challenge.leadOrganisations || [];
+
+  const available = useMemo(() => organisations.filter(x => !leadingOrganisations.find(y => y.id === x.id)), [
+    organisations,
+    leadingOrganisations,
+  ]);
+  const leading = useMemo(() => organisations.filter(x => leadingOrganisations.find(y => y.id === x.id)), [
+    organisations,
+    leadingOrganisations,
+  ]);
+
+  const [updateChallenge, { loading: isUpdating }] = useUpdateChallengeMutation({
+    onError: handleError,
+    refetchQueries: [refetchChallengeLeadOrganisationsQuery({ ecoverseId: ecoverseId, challengeID: challengeId })],
+    awaitRefetchQueries: true,
+  });
+
+  const handleAdd = (orgId: string) => {
+    const newLeading = [...leadingOrganisations.map(x => x.id), orgId];
+    _updateChallenge({
+      ID: challengeId,
+      leadOrganisations: newLeading,
+    });
+  };
+
+  const handleRemove = (orgId: string) => {
+    const orgToRemoveIndex = leadingOrganisations.findIndex(x => x.id === orgId);
+
+    if (orgToRemoveIndex > -1) {
+      const newLeading = [...leadingOrganisations].map(x => x.id);
+      newLeading.splice(orgToRemoveIndex, 1);
+
+      _updateChallenge({
+        ID: challengeId,
+        leadOrganisations: newLeading,
+      });
+    }
+  };
+
+  const _updateChallenge = (input: UpdateChallengeInput) => {
+    updateChallenge({
+      variables: {
+        input: {
+          ID: input.ID,
+          leadOrganisations: input.leadOrganisations,
+        },
+      },
+    });
+  };
+
+  // todo type
+  return (
+    <EditLeadingOrganisation
+      available={available as any}
+      leading={leading as any}
+      onAdd={handleAdd}
+      onRemove={handleRemove}
+      isUpdating={isUpdating}
+    />
+  );
+};
+
+interface Type {
```

Comment:
This type must be the same as the return type of the query - here I am using a substitue type.
How can this be achieved @atstojanov ?